src/OpenTelemetry.AutoInstrumentation.Native/continuous_profiler.cpp

```diff
@@ -210,28 +209,24 @@ int32_t AllocationSamplingConsumeAndReplaceBuffer(int32_t len, unsigned char* bu
     return static_cast<int32_t>(to_use_len);
 }
 
-// TODO: deduplicate
-int32_t SelectiveSamplingConsumeAndReplaceBuffer(int32_t len, unsigned char* buf)
+static int32_t SelectiveSamplingConsumeAndClearBuffer(int32_t len, unsigned char* buf)
 {
     if (len <= 0 || buf == nullptr)
     {
-        trace::Logger::Warn("Unexpected 0/null buffer to SelectiveSamplingConsumeAndReplaceBuffer");
+        trace::Logger::Warn("Unexpected 0/null buffer to SelectiveSamplingConsumeAndClearBuffer");
         return 0;
     }
-    std::vector<unsigned char>* to_use = nullptr;
-    {
-        std::lock_guard<std::mutex> guard(selective_sampling_buffer_lock);
-        to_use                    = selective_sampling_buffer;
-        selective_sampling_buffer = new std::vector<unsigned char>();
-        selective_sampling_buffer->reserve(kSamplesBufferDefaultSize);
-    }
-    if (to_use == nullptr)
+
+    std::lock_guard<std::mutex> guard(selective_sampling_buffer_lock);
+    const size_t                to_use_len = std::min(selective_sampling_buffer.size(), static_cast<size_t>(len));
+
+    if (to_use_len == 0)
     {
         return 0;
     }
-    const size_t to_use_len = static_cast<int>(std::min(to_use->size(), static_cast<size_t>(len)));
-    memcpy(buf, to_use->data(), to_use_len);
-    delete to_use;
+    memcpy(buf, selective_sampling_buffer.data(), to_use_len);
```

Comment:
This is on par with what the behavior was before this change. So, not a blocker for this PR. Nevertheless, I am mentioning my observation - 
if "to_use_len" was smaller than sampling buffer size, we throw away the remainder of the data. Something to consider in future.